tensorzero-core/src/inference/types/pyo3_helpers.rs

```diff
@@ -390,10 +395,11 @@ pub fn deserialize_from_stored_sample<'a>(
             Datapoint::Json(json_wire) => Ok(StoredSampleItem::Datapoint(StoredDatapoint::Json(
                 json_wire,
             ))),
-        }
-    } else {
-        deserialize_from_pyobj(py, obj)
+        };
     }
+
+    // Fall back to generic deserialization
+    deserialize_from_pyobj(py, obj)
```

Comment:
for my own understanding: we need these fallbacks because there is still a possibility that the user has passed a PyO3 type that we know how to accept and we need to directly extract those before trying the more standard deserialization-based approach

Reply:
makes sense to me